internal/provider/resource_cluster_personality.go

```diff
@@ -0,0 +1,145 @@
+// Copyright 2024 Broadcom. All Rights Reserved.
+// SPDX-License-Identifier: MPL-2.0
+
+package provider
+
+import (
+	"context"
+	"fmt"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"github.com/vmware/terraform-provider-vcf/internal/api_client"
+	"github.com/vmware/vcf-sdk-go/client/personalities"
+	"github.com/vmware/vcf-sdk-go/client/vcenters"
+	"github.com/vmware/vcf-sdk-go/models"
+	"time"
+)
+
+const (
+	uploadModeReferred = "REFERRED"
+)
+
+func ResourceClusterPersonality() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceClusterPersonalityCreate,
+		ReadContext:   resourceClusterPersonalityRead,
+		DeleteContext: resourceClusterPersonalityDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(30 * time.Minute),
+			Update: schema.DefaultTimeout(30 * time.Minute),
+			Delete: schema.DefaultTimeout(30 * time.Minute),
+		},
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				Description:  "The name for the personality",
+				ValidateFunc: validation.NoZeroValues,
+				ForceNew:     true,
+			},
+			"domain_id": {
+				Type:         schema.TypeString,
+				Required:     true,
+				Description:  "The identifier of the domain which contains the vcenter and source cluster",
+				ValidateFunc: validation.NoZeroValues,
+				ForceNew:     true,
+			},
+			"cluster_id": {
+				Type:         schema.TypeString,
+				Required:     true,
+				Description:  "The identifier of the source cluster within the vCenter server (e.g. domain-c1)",
+				ValidateFunc: validation.NoZeroValues,
+				ForceNew:     true,
+			},
+		},
+	}
+}
+
+func resourceClusterPersonalityCreate(ctx context.Context, data *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(*api_client.SddcManagerClient).ApiClient
+
+	mode := uploadModeReferred
+	name := data.Get("name").(string)
+
+	vcenterId, err := getVcenterId(data, meta)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	clusterId := data.Get("cluster_id").(string)
+
+	spec := models.PersonalityUploadSpec{
+		Name:       name,
+		UploadMode: &mode,
```

Comment:
Is it possible to use a different mode? If not, then maybe we should mention this in the docs.

Reply:
It's explained in the PR description. I'll add it to the docs too.